From 4e2925132ae1613d5304d136d6985fcabab28d17 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:45 -0400
Subject: [fs] nfsd: fix lost nfserrno() call in nfsd_setattr()

Message-id: <1403560314-32525-44-git-send-email-bfields@redhat.com>
Patchwork-id: 84474
O-Subject: [Patch RHEL7 043/172] nfsd: fix lost nfserrno() call in nfsd_setattr()
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: "J. R. Okajima" <hooanon05g@gmail.com>

There is a regression in
 208d0ac 2014-01-07 nfsd4: break only delegations when appropriate
which deletes an nfserrno() call in nfsd_setattr() (by accident,
probably), and NFSD becomes ignoring an error from VFS.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 1406b916f4a29d5f9660264a28ce609c8c77e7ae
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/vfs.c b/fs/nfsd/vfs.c
index ba972a1..aba35f1 100644
--- a/fs/nfsd/vfs.c
+++ b/fs/nfsd/vfs.c
@@ -449,6 +449,7 @@ nfsd_setattr(struct svc_rqst *rqstp, struct svc_fh *fhp, struct iattr *iap,
  fh_lock(fhp);
  host_err = notify_change(dentry, iap, NULL);
  fh_unlock(fhp);
+ err = nfserrno(host_err);
 
 out_put_write_access:
  if (size_change)
-- 
1.7.1